lib/src/signed_video_openssl.c

```diff
@@ -522,3 +524,28 @@ signed_video_generate_private_key(sign_algo_t algo,
 
   return svi_rc_to_signed_video_rc(status);
 }
+
+svi_rc
+check_type_of_key(signed_video_t *self, const char *public_key, size_t public_key_size)
```

Comment:
I don't think it is necessary to take the entire `signed_video_t`.
Better to stick with `sign_algo_t`, and put it as las argument since it is an output.
Also, the name is misleading since it does not only check, but also set algo. Further, the name should be prefixed with `openssl`. So for example,
```
openssl_get_algo_of_public_key(const char *public_key, size_t public_key_size, sign_algo_t *algo)
```